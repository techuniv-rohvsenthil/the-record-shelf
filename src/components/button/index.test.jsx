import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './index';


describe('the button component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button testID="test-btn" click={() => {}}>Test text</Button>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call the onClick prop when the button is clicked', () => {
    const onClickMock = jest.fn();
    const { getByTestId } = render(<Button testID="test-btn" click={onClickMock}>Test text</Button>);

    fireEvent.click(getByTestId('test-btn'));
    expect(onClickMock).toHaveBeenCalled();
  });
});
