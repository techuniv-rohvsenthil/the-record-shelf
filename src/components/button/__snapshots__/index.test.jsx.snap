// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button component should render correctly 1`] = `
<DocumentFragment>
  <button
    data-testid="test-btn"
    type="button"
  >
    Test text
  </button>
</DocumentFragment>
`;
